tools/loader/ImageClassifier.cpp

```diff
@@ -101,6 +112,24 @@ static bool getNextImageFilenames(std::vector<std::string> *filenames) {
   return !filenames->empty();
 }
 
+/// Generate in \p imageList the list of filenames corresponding to the next
+/// mini-batch of size \p miniBatchSize extracted from \p totalImageList at
+/// index \p minibatchIndex.
+static bool getNextMiniBatch(std::vector<std::string> &imageList,
```

Comment:
nit: Maybe could use `llvm::MutableArrayRef<std::string> imageList` here. Though not sure there's a `clear()` for it.

Reply:
`llvm::MutableArrayRef` is quite limited: no clear, not possibile to add an element too it seems. I then kept `std::vector<std::string> &`